compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1328,11 +1361,13 @@ rustc_queries! {
 
     query impl_defaultness(def_id: DefId) -> hir::Defaultness {
         desc { |tcx| "looking up whether `{}` is a default impl", tcx.def_path_str(def_id) }
+        cache_on_disk_if { def_id.is_local() }
         separate_provide_extern
     }
 
     query impl_constness(def_id: DefId) -> hir::Constness {
         desc { |tcx| "looking up whether `{}` is a const impl", tcx.def_path_str(def_id) }
+        cache_on_disk_if { def_id.is_local() }
```

Comment:
Aren't those just a cheap hir lookup?

Reply:
Yes. For the moment, I'm trying some heuristic like "cache everything that will be stored in metadata (roughly if is has separate_provide_extern)". I plan to refine this from perf measurements.